src/v2/Components/FlashBanner/EmailConfirmationCTA.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react"
+import { commitMutation, graphql } from "react-relay"
+import { Button, Sans } from "@artsy/palette"
+import { EmailConfirmationCTAMutationResponse } from "v2/__generated__/EmailConfirmationCTAMutation.graphql"
+import { useSystemContext } from "v2/Artsy"
+
+export const EmailConfirmationCTA: React.FC = () => {
+  const [afterSubmitContent, setAfterSubmitContent] = React.useState<string>(
+    null
+  )
+  const { relayEnvironment } = useSystemContext()
+
+  const requestConfirmation = () => {
+    return new Promise<EmailConfirmationCTAMutationResponse>((done, reject) => {
+      commitMutation(relayEnvironment, {
+        onCompleted: (data, errors) => {
+          console.log("completed")
+
+          errors && errors.length ? reject(errors) : done(data)
+        },
+        onError: error => {
+          console.log("error", error)
+          reject(error)
+        },
+        mutation: graphql`
+          mutation EmailConfirmationCTAMutation {
+            sendConfirmationEmail(input: {}) {
+              confirmationOrError {
+                ... on SendConfirmationEmailMutationSuccess {
+                  unconfirmedEmail
+                }
+                ... on SendConfirmationEmailMutationFailure {
+                  mutationError {
+                    error
+                    message
+                  }
+                }
+              }
+            }
+          }
+        `,
+      })
+    })
+  }
+
+  const handleSubmit = async () => {
```

Comment:
If going to mark as `async` would be good to switch from promises internally to `async / await`

Reply:
youre right, i dont think the return type even matters here.